docs/tutorials/runtime.md

```diff
@@ -0,0 +1,265 @@
+- [FAQ](#faq)
+
+<!-- TOC -->
+
+## Customize Workflow
+
+By default, we recommend users to use **`EvaluationHook`** to do evaluation after training epoch, but they can still use `val` workflow as an alternative.
+
+Workflow is a list of (phase, epochs) to specify the running order and epochs. By default it is set to be
+
+```python
+workflow = [('train', 1)]
+```
+
+which means running 1 epoch for training.
+Sometimes user may want to check some metrics (e.g. loss, accuracy) about the model on the validate set.
+In such case, we can set the workflow as
+
+```python
+[('train', 1), ('val', 1)]
+```
+
+so that 1 epoch for training and 1 epoch for validation will be run iteratively.
+
+```{note}
+1. The parameters of model will not be updated during val epoch.
+2. Keyword `total_epochs` in the config only controls the number of training epochs and will not affect the validation workflow.
+3. Workflows `[('train', 1), ('val', 1)]` and `[('train', 1)]` will not change the behavior of `EvalHook` because `EvalHook` is called by `after_train_epoch` and validation workflow only affect hooks that are called through `after_val_epoch`.
+   Therefore, the only difference between `[('train', 1), ('val', 1)]` and ``[('train', 1)]`` is that the runner will calculate losses on validation set after each training epoch.
+```
+
+## Hooks
+
+The hook mechanism is widely used in the OpenMMLab open source algorithm library. Combined with the `Runner`, the entire life cycle of the training process can be managed easily. You can learn more about the hook through [related article](https://www.calltutors.com/blog/what-is-hook/).
+
+Hooks only work when they are registered in the constructor. At present, hooks are mainly divided into two categories:
+
+- default runtime hooks
+
+The default runtime hooks are registered by the runner by default. Generally, they are hooks for some basic functions, and have a certain priority, there is no need to modify the priority.
+
+- non-default runtime hooks
+
+The non-default runtime hooks are registered through `custom_hooks`. Generally, they are hooks with enhanced functions. The priority needs to be specified in the configuration file. If you do not specify the priority of the hook, it will be set to'NORMAL' by default.
+
+**Priority list**
+
+| Level           | Value      |
+|:--:|:--:|
+| HIGHEST         | 0          |
+| VERY_HIGH       | 10         |
+| HIGH            | 30         |
+| ABOVE_NORMAL    | 40         |
+| NORMAL(default) | 50         |
+| BELOW_NORMAL    | 60         |
+| LOW             | 70         |
+| VERY_LOW        | 90         |
+| LOWEST          | 100        |
+
+The priority determines the execution order of the hooks. Before training, the log will print out the execution order of the hooks at each stage to facilitate debugging.
+
+### Default runtime hooks
+
+There are some common hooks that are not registered through `custom_hooks` but has been registered by default when importing MMCV, they are
```

Comment:
The default training hooks are not registered when importing MMCV, they're registered in `mmcls/apis/train.py`